zerver/lib/actions.py

```diff
@@ -4578,8 +4638,31 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
             # Remove duplicates by excluding the id of users already in users_to_be_notified list.
             # This is the case where a user both has a UserMessage row and is a current Subscriber
             subscribers = subscribers.exclude(user_profile_id__in=[um.user_profile_id for um in ums])
+
+            if new_stream is not None:
+                assert guest_subs_losing_access is not None
+                # Exclude guest users who are not subscribed to the new stream from receing this event.
+                subscribers = subscribers.exclude(user_profile_id__in=[sub.user_profile_id for sub in guest_subs_losing_access])
+
             # All users that are subscribed to the stream must be notified when a message is edited
             subscribers_ids = [user.user_profile_id for user in subscribers]
+
+            if new_stream is not None:
+                # TODO: Guest users don't see the new moved topic unless breadcrumb message for
+                # new stream is enabled. Excluding these users from receiving this event helps
+                # us avoid a error trackeback for our clients. We should figure out a way to
+                # inform the guest users of this new topic if sending a 'message' event for these messages
+                # is not an option.
+                # Don't send this event to guest subs who are not subscrbied to the old stream but
+                # are subscribed to the new stream
```

Comment:
What are the details on the traceback?  I'm wondering if there's a general client-side bug when receiving message_update events for a message that isn't cached in the browser.

Reply:
(Or is it just for cases where the old stream isn't known?  Because that we can probably address client-side)

Reply:
Client doesn't know about the old stream, I don't think we can always expect clients to handle this case for us. I can see unnoticed regressions happening in the future. Clients also don't have the moved messages stored locally, so they will have to request server for info anyway, which seems like a wasted round trip. There are some advantages I see doing it on the server side:

* We have all the info about subscribers and messages on server, so we can decide easily which event to send to which client. So, client doesn't need to receive an event which it clearly cannot handle, and we're deliberately sending an event to the client just for them to ignore, it doesn't make sense.

* On server, we can directly send say a "message" event for these messages in future which client handles gracefully in this case.

Reply:
Well, we'd still be asking the client to understand a `message` event for messages that aren't new; I'm pretty sure the Zulip webapp certainly doesn't support that as a general rule.  (See the `messages_are_new` flag)